Supervised_learning/resnet_lib.py

```diff
@@ -0,0 +1,88 @@
+import tensorflow as tf
+from tensorflow.keras.applications.resnet50 import ResNet50
+from tensorflow.keras.preprocessing import image
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers import Dense, GlobalAveragePooling2D, Input, Dropout
+from tensorflow.keras.callbacks import Callback
+from tensorflow.keras.optimizers import SGD, RMSprop
+import numpy as np
+
+tf.enable_eager_execution()
+
+# Read in tfrecord files and construct dataset from it
+def create_dataset(filepath):
+    raw_image_dataset = tf.data.TFRecordDataset(filepath)
+
+    # Create a dictionary describing the features.
+    image_feature_description = {
+        'label': tf.io.FixedLenFeature([], tf.int64),
+        'image': tf.io.FixedLenFeature([], tf.string),
+    }
+
+    def _parse_image_function(example_proto):
+      # Parse the input tf.Example proto using the dictionary above.
+      return tf.io.parse_single_example(example_proto, image_feature_description)
+
+    parsed_image_dataset = raw_image_dataset.map(_parse_image_function)
+    
+    image, label = [], []
+    
+    for image_features in parsed_image_dataset:
+        label.append(image_features['label'])
+        image.append(tf.image.decode_image(image_features['image']).numpy())
+    
+    return np.asarray(image), np.asarray(label)
+
+
+# Callback function to check test set accuracy and loss for every training epoch.
+class test_callback(Callback):
+    def __init__(self, test_data):
+        self.test_data = test_data
+
+    def on_epoch_end(self, epoch, logs={}):
+        image, label = self.test_data
+        loss, acc = self.model.evaluate(image, label, verbose=0)
+        print('\nTesting loss: {}, acc: {}\n'.format(loss, acc))
+
+
+# Create transfer learning model from pretrained Resnet50 model.
+def create_model(image_size, num_classes, lr):
```

Comment:
Can you add comments to the function header to explain the input/output types? I know some of these are straightforward, but often with Python it can be hard to tell what "type" you are working with---especially when reading someone else's code.

Reply:
Done. I've added comments for the input/output types for all functions.
I do not have access to go/pystyle, but there is a google python style guide here: https://google.github.io/styleguide/pyguide.html